aea/test_tools/test_contract.py

```diff
@@ -39,11 +39,52 @@
 )
 
 
-class BaseContractTestCase(ABC):
+class _MetaBaseContractTestCase(ABCMeta):
+    """A metaclass that validates BaseContractTestCase's class."""
+
+    def __new__(  # type: ignore # pylint: disable=bad-mcs-classmethod-argument # type: ignore
+        mcs, name: str, bases: Tuple, namespace: Dict, **kwargs: Any
+    ) -> Type:
+        """Initialize the class."""
+        new_cls = super().__new__(mcs, name, bases, namespace, **kwargs)
+
+        if ABC in bases:
+            # abstract class, return
+            return new_cls
+        if not issubclass(new_cls, BaseContractTestCase):
+            # the check only applies to BaseContractTestCase subclasses
+            return new_cls
+
+        mcs._check_consistency(cast(Type[BaseContractTestCase], new_cls))
+        return new_cls
+
+    @classmethod
+    def _check_consistency(  # pylint: disable=bad-mcs-classmethod-argument
+        mcs, base_contract_test_case_cls: Type["BaseContractTestCase"]
+    ) -> None:
+        """Check consistency of class."""
+        mcs._check_required_class_attributes(base_contract_test_case_cls)
+
+    @classmethod
+    def _check_required_class_attributes(  # pylint: disable=bad-mcs-classmethod-argument
+        mcs, base_contract_test_case_cls: Type["BaseContractTestCase"]
+    ) -> None:
+        """Check that required class attributes are set."""
+        if not hasattr(base_contract_test_case_cls, "path_to_contract"):
+            raise ValueError(
+                f"'path_to_contract' not set on {base_contract_test_case_cls}"
+            )
+        if not hasattr(base_contract_test_case_cls, "ledger_identifier"):
+            raise ValueError(
+                f"'ledger_identifier' not set on {base_contract_test_case_cls}"
+            )
+
+
+class BaseContractTestCase(ABC, metaclass=_MetaBaseContractTestCase):
     """A class to test a contract."""
 
-    path_to_contract: Path = Path(".")
-    ledger_identifier: str = ""
+    path_to_contract: Path
+    ledger_identifier: str
```

Comment:
Setting default values here is undesirable: https://github.com/valory-xyz/open-aea/issues/492

Now we also validate via the meta class